core_api/src/routes/file.py

```diff
@@ -0,0 +1,131 @@
+import logging
+from uuid import UUID
+
+from fastapi import FastAPI, HTTPException
+from faststream.redis.fastapi import RedisRouter
+from pydantic import AnyHttpUrl
+
+from redbox.models import Chunk, File, FileStatus, Settings
+from redbox.storage import ElasticsearchStorageHandler
+
+# === Logging ===
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger()
+
+env = Settings()
+
+
+# === Object Store ===
+
+s3 = env.s3_client()
+
+
+# === Queues ===
+router = RedisRouter(url=env.redis_url)
+publisher = router.publisher(env.ingest_queue_name)
+
+
+# === Storage ===
+
+es = env.elasticsearch_client()
+storage_handler = ElasticsearchStorageHandler(es_client=es, root_index="redbox-data")
+
+
+file_app = FastAPI(
+    title="Core File API",
+    description="Redbox Core File API",
+    version="0.1.0",
+    openapi_tags=[
+        {"name": "file", "description": "File endpoints"},
+    ],
+    docs_url="/docs",
+    redoc_url="/redoc",
+    openapi_url="/openapi.json",
+    lifespan=router.lifespan_context,
+)
+file_app.include_router(router)
+
+
+@file_app.post("/", tags=["file"])
+async def create_upload_file(name: str, type: str, location: AnyHttpUrl) -> UUID:
+    """Upload a file to the object store and create a record in the database
+
+    Args:
+        name (str): The file name to be recorded
+        type (str): The file type to be recorded
+        location (AnyHttpUrl): The presigned file resource location
+
+    Returns:
+        UUID: The file uuid from the elastic database
+    """
+
+    file = File(
+        name=name,
+        url=str(location),  # avoids JSON serialisation error
+        content_type=type,
+    )
+
+    storage_handler.write_item(file)
+
+    log.info(f"publishing {file.uuid}")
+    await router.broker.connect()
```

Comment:
does this need to happen once as part of setup, or even within https://fastapi.tiangolo.com/advanced/events/?

Reply:
I think moving it above would be a lot cleaner network IO wise

Reply:
Unfortunately, the setup only happens on the root app, not any sub ones (https://github.com/tiangolo/fastapi/issues/1480#issuecomment-636317883 )

Reply:
@lmwilkigov after speaking to @rachaelcodes I see that this is not straightforward. I am going to approve this and raise a tech debt ticket to come back to this.

Reply:
https://technologyprogramme.atlassian.net/browse/REDBOX-154